test/unit/org/apache/cassandra/index/sai/cql/AllowFilteringTest.java

```diff
@@ -514,4 +518,31 @@ public void testMapRangeQueries()
         // Show that we're now able to execute the query.
         assertRows(execute("SELECT partition FROM %s WHERE item_cost['apple'] < 3 AND item_cost['apple'] > 1"), row(0));
     }
+
+    private Injections.Barrier blockIndexBuild = Injections.newBarrier("block_index_build", 2, false)
+            .add(InvokePointBuilder.newInvokePoint().onClass(StorageAttachedIndex.class)
+                    .onMethod("startInitialBuild"))
+            .build();
+
+    @Test
+    public void testAllowFilteringDuringIndexBuild() throws Throwable
+    {
+        createTable("CREATE TABLE %s (k int PRIMARY KEY, v int)");
+        Injections.inject(blockIndexBuild);
+        String idx = createIndexAsync(String.format("CREATE CUSTOM INDEX ON %%s(v) USING '%s'", StorageAttachedIndex.class.getName()));
+
+        assertThatThrownBy(() -> execute("SELECT * FROM %s WHERE v=0"))
+                .hasMessage("The secondary index '" + idx + "' is not yet available as it is building")
+                .isInstanceOf(IndexBuildInProgressException.class);
+
+        assertThatThrownBy(() -> execute("SELECT * FROM %s WHERE v=0 ALLOW FILTERING"))
+                .hasMessage("The secondary index '" + idx + "' is not yet available as it is building")
+                .isInstanceOf(IndexBuildInProgressException.class);
```

Comment:
Why try to use the index and fail  instead of use filtering if the user allowed it? 
I thought if I set AF in the query, the query should always run regardless of index availability.

Reply:
The AF issue is a different ticket where we are addressing this - #12425. Until that one is solved, we add as an intermediate this  one as per agreement with interested parties.